pkgs/development/python-modules/cirq-rigetti/default.nix

```diff
@@ -1,46 +1,41 @@
 {
+  python,
+  fetchPypi,
   buildPythonPackage,
-  cirq-core,
   setuptools,
-  pyquil,
-  qcs-sdk-python,
-  pytestCheckHook,
+  numpy_1,
+  cirq-core,
 }:
 
+# Cirq-Rigetti is deprecated and limited to python 3.12 and Numpy 1
+# per https://github.com/quantumlib/Cirq/releases/tag/v1.5.0
+# For more details or to provide feedback see https://github.com/quantumlib/Cirq/issues/7058
+
 buildPythonPackage rec {
   pname = "cirq-rigetti";
-  pyproject = true;
-  inherit (cirq-core) version src;
-
-  sourceRoot = "${src.name}/${pname}";
+  version = "1.5.0";
+  format = "wheel";
 
-  pythonRelaxDeps = [ "pyquil" ];
+  disabled = (python.pythonVersion != "3.12");
 
-  postPatch = ''
-    # Remove outdated test
-    rm cirq_rigetti/service_test.py
-  '';
+  src = fetchPypi {
+    inherit version format;
+    pname = "cirq_rigetti";
+    dist = "py3";
+    python = "py3";
+    hash = "sha256-DIe0fsQSrIo9xnOmk0bO/v6MZsEQhBY36cT7xfSd8nA=";
+  };
 
   build-system = [ setuptools ];
 
   dependencies = [
-    cirq-core
-    pyquil
-    qcs-sdk-python
-  ];
-
-  nativeCheckInputs = [ pytestCheckHook ];
-
-  disabledTestPaths = [
-    # No need to test the version number
-    "cirq_rigetti/_version_test.py"
+    numpy_1
```

Comment:
Python packages can on;y use `numpy`.

Reply:
From the release notes: 
> Cirq-Rigetti needs to be installed explicitly using pip install cirq-rigetti as it is not included with the cirq installation anymore. Note that installation of Cirq-Rigetti requires a downgrade to Python 3.12 and NumPy-1.

Which is why I'm doing this wierdly regressive thing of building from PyPi instead of source. They're decoupling -regetti from the cirq-core lifecycle.

If there's a better way to do this, I'd love to hear it.

Reply:
If necessary, use `fetchFromGitHub` directly rather than inheriting `src`.

Reply:
Based on https://github.com/quantumlib/Cirq/blob/main/cirq-rigetti/cirq_rigetti/deprecation.py, the actual deprecation deadline is 1.6.0. So I'm backing out to building this component as usual.